src/hooks.ts

```diff
@@ -36,3 +43,112 @@ export const useHasResourcePermission = (resource: Resource, authzUrl: string, p
     const { permissions, isFetching } = useResourcePermissions(resource, authzUrl) ?? {};
     return { isFetching, hasPermission: permissions.includes(permission) };
 };
+
+export const useOpenIdConfig = (openIdConfigUrl: string) => {
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        dispatch(fetchOpenIdConfiguration(openIdConfigUrl));
```

Comment:
I think this effect results in a call every time the hook is used, even if openIdConfiguration is already loaded, and if this hook is used more than once I think it'll dispatch multiple times each time the URL changes - I think separating the effect into a separate fetchIfNecessary hook and keeping this to just a useSelector would be good...